app/src/main/java/ch/epfl/favo/view/tabs/addFavor/FavorRequestView.java

```diff
@@ -231,6 +232,16 @@ private void setupButtons(View rootView) {
         });
   }
 
+  private void deleteFavor() {
+    CompletableFuture deleteFuture = getViewModel().deleteFavor(currentFavor);
+    deleteFuture.thenAccept(
+        o -> {
+          CommonTools.showSnackbar(requireView(), getString(R.string.favor_delete_success_msg));
+          requireActivity().onBackPressed();
+        });
+    deleteFuture.exceptionally(onFailedResult(requireView()));
+  }
+
   /**
```

Comment:
are there any scenarios in which currentFavor could be nil? or getViewModel() returns nil? im not too sure myself so im asking

Reply:
getViewModel() no because that is handled by the ViewLifeCycleController, which will always be non-null if the view is active. 

currentFavor could be null if we set it to be null in the FavorViewModel. If the favor object is deleted from the DB the observed query will throw an exception instead of returning a null value, which will just be caught in a snackbar on the view. 

If we don't delete the observer, then the snackbar will appear, that's why I just trigger onBackPressed() as soon as we delete a favor instead of manually removing the observer from the LiveData object that does not exist anymore.